test/libsolidity/semanticTests/functionSelector/function_selector_as_constant.sol

```diff
@@ -0,0 +1,11 @@
+contract B {
+    function g() public {}
+}
+
+contract C is B {
+    bytes4 public constant s2 = B.g.selector;
```

Comment:
We should add some coverage for related cases:
- Using selector in other places outside of functions. E.g.:
    - constructor args when called as a modifier
    - constructor args when called after `is`
    - immutable initialization
    - state variable initialization
- Using the reference to the function itself (but still without calling it).